Mathlib/Topology/EMetricSpace/Arclength.lean

```diff
@@ -0,0 +1,378 @@
+/-
+Copyright (c) 2023 Junyan Xu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junyan Xu, Success Moses
+-/
+module
+
+public import Mathlib.Topology.EMetricSpace.BoundedVariation
+public import Mathlib.Data.Set.Operations
+
+/-!
+# Arc length
+
+This file is devoted to the definition of the `arcLength` of a function `f` between
+`a` and `b` and the continuity of the single variable function `arcLength a` or
+`(arcLength · b)`.
+
+## Tags
+
+Topology, Metric space, Continuity
+-/
+
+@[expose] public section
+
+open ENNReal
+
+variable {α E : Type*} [LinearOrder α] [PseudoEMetricSpace E] (f : α → E) {a b c : α}
+
+/-- The length of the arc of the curve `f : α → E` between two points `a` and `b` is defined
+as the variation of `f` on the closed interval `[a, b]`. Equals zero when `b ≤ a`. -/
+noncomputable def arcLength (a b : α) : ℝ≥0∞ :=
+  eVariationOn f (Set.Icc a b)
+
+theorem arcLength_eq : arcLength f a b = eVariationOn f (Set.Icc a b) := rfl
+
+theorem arcLength_ne_iff : (arcLength f a b ≠ ∞) ↔ BoundedVariationOn f (Set.Icc a b) := by rfl
```

Comment:
```suggestion
theorem arcLength_ne_top_iff : arcLength f a b ≠ ∞ ↔ BoundedVariationOn f (Set.Icc a b) := .rfl
```
I think we should still use `arcLength f a b ≠ ∞` as the default spelling, and rewrite with this when needed.

Or possibly introduce an abbrev `RectifiableOn f a b := arcLength f a b ≠ ∞` ...